internal/workflow/state.go

```diff
@@ -80,26 +78,19 @@ type sipInfo struct {
 	// isDir indicates whether the working copy of the SIP is a directory.
 	isDir bool
 
+	// extension is the original SIP file extension if it was not a directory.
+	// Used to generate the failed SIP/PIP object key.
+	extension string
+
 	// status is the ingest status of the SIP (e.g. "in progress", "done",
 	// "error").
 	status enums.SIPStatus
-}
-
-// pipInfo represents the PIP.
-type pipInfo struct {
-	// id is the UUID of the PIP. It is populated by the
-	// am.PollTransferActivity.
-	id string
-
-	// path is the path to the PIP. It is populated by the BundleActivity.
-	path string
 
-	// isDir indicates whether the current working copy of the SIP is a
-	// filesystem directory.
-	isDir bool
+	// sipType is the type of the SIP.
+	sipType enums.SIPType
 
-	// pipType is the type of the PIP.
-	pipType enums.SIPType
+	// transformed indicates if the SIP should be considered as a PIP.
```

Comment:
The "as" is unnecessary: "transformed indicates if the SIP should be considered a PIP."